### PR TITLE
FirefoxCI cluster upgrade docs

### PR DESCRIPTION
**Pull request by escapewindow**
Initial pass. We can update these as the process changes.

### CODE REVIEW

File: taskcluster/fxci_upgrades.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+.. _major_tc_upgrade:
+
+Major version upgrade process
+-----------------------------
+
+If there is a major version upgrade, or other maintenance/migration that will have larger side effects than a few busted tasks in a half hour of maintenance, let's follow these steps:
+
+- Identify the potential issues we might hit post-upgrade/maintenance/migration
+- :ref:`Test in taskcluster staging <tc_staging>`
```
Comment (matt-boris) [inline taskcluster/fxci_upgrades.rst:43]: This creates a broken link

File: taskcluster/tc_staging.rst [hunk 2]
```diff
@@ -77,6 +84,29 @@ You'll need :ref:`taskcluster_cli` for this.
...
+    secrets:get:project/releng/gecko/build/level-3/gecko-generated-sources-upload
+    secrets:get:project/releng/gecko/build/level-3/gecko-symbol-upload
+    secrets:get:project/taskcluster/gecko/hgfingerprint
+    secrets:get:project/taskcluster/gecko/hgmointernal
+
+It's possible we just need the ``gecko/build`` and ``taskcluster/gecko/hg*`` secrets.
+
+`This script <https://hg.mozilla.org/build/braindump/file/tip/taskcluster/copy_secrets_to_staging.py>`__ copies that subset of secrets from fxci to staging. We need to do the following to use it::
+
+- set the ``NOOP`` boolean to ``False`` in the script
```
Comment (matt-boris) [inline taskcluster/tc_staging.rst:108]: Screen Shot 2022-04-13 at 8 45 47 PM

Unintended code-block styling?

File: taskcluster/fxci_upgrades.rst (hunk 1 above)
Comment (escapewindow) [reply taskcluster/fxci_upgrades.rst:43]: I think it's the difference between plain .rst and [sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html?highlight=ref#role-ref).
In the [PR RtD build](https://moz-releng-docs--154.org.readthedocs.build/en/154/taskcluster/fxci_upgrades.html#major-version-upgrade-process), the link shows up and goes [here](https://moz-releng-docs--154.org.readthedocs.build/en/154/taskcluster/tc_staging.html#tc-staging).

File: taskcluster/fxci_upgrades.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
+FirefoxCI cluster upgrades
+==========================
+
+As of H1 2022, the Taskcluster, Cloudops, and Releng teams have committed to rolling out upgrades to the FirefoxCI cluster at a regular cadence.
+
+Process
+-------
+
+We now have a regular cadence of scheduled upgrades, as noted in the `Public - FirefoxCI Cluster Taskcluster Upgrades calendar`_.
```
Comment (hwine) [inline taskcluster/fxci_upgrades.rst:9]: nit: calendar URL doesn't load that calendar

Comment (escapewindow) [reply taskcluster/fxci_upgrades.rst:9]: interesting, it brings me to my calendar and says i've loaded it already.
i've updated the link to a calendar view.

Comment (hwine) [reply taskcluster/fxci_upgrades.rst:9]: nit part deux 😀: The calendar title says "Public", but the link appears to be only available to company accounts. (I'm asked to auth when attempting to view in a private window.)

File: taskcluster/fxci_upgrades.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- Close trees 2+hours before (Sheriffs or Releaseduty)
+- Check with Relman before proceeding (Releaseduty)
+- Roll out the maintenance fixes and cluster upgrades (Cloudops team)
+- Check on smoketests (Taskcluster team)
+- Check treeherder and ask Sheriffs if there are any broken tasks (Releaseduty)
+- Reopen trees (Sheriffs or Releaseduty)
+- Send an email saying the upgrade is finished (Releaseduty)
+- Update the `changelog`_.
+
+.. _Public - FirefoxCI Cluster Taskcluster Upgrades calendar: https://calendar.google.com/calendar/embed?src=c_fmd7bfpgr3929rrmiejbk3us68%40group.calendar.google.com
```
Comment (hwine) [inline taskcluster/fxci_upgrades.rst:56]: nit part deux 😀: The calendar title says "Public", but the link appears to be only available to company accounts. (I'm asked to auth when attempting to view in a private window.)

File: taskcluster/fxci_upgrades.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+- We remind everyone about these upgrades by sending out emails like `this <https://groups.google.com/a/mozilla.org/g/dev-platform/c/UGCNchYxVns>`__, and checking with `Relman <https://wiki.mozilla.org/Release_Management/Release_owners>`__ before the upgrade to make sure there are no releases in-flight.
+
+.. _minor_tc_upgrade:
+
+Minor version upgrade process
+-----------------------------
+
+As of 2022.04.13, we have been successful rolling out minor version Taskcluster upgrades *without a tree closure*. We also managed to upgrade the database instance RAM during such a non-tree-closure window, which made the DB unavailable for ~10-15min. This only resulted in ~2 failed tasks, which went green on rerun. So if we want to roll out a minor version upgrade or other short-term outage maintenance tasks:
+
+- Send an email noting that this is a tree-closure-less upgrade several days before (Releaseduty)
```
Comment (gbrownmozilla) [inline taskcluster/fxci_upgrades.rst:28]: Maybe specify what groups the email should be sent to?

File: taskcluster/fxci_upgrades.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+- Check with Relman before proceeding (Releaseduty)
```
Comment (gbrownmozilla) [inline taskcluster/fxci_upgrades.rst:29]: Maybe specify where to check - #releaseduty?

**Review by gbrownmozilla — APPROVED** [review]

File: taskcluster/fxci_upgrades.rst (hunk 4 above)
Comment (escapewindow) [reply taskcluster/fxci_upgrades.rst:56]: This was intentional, but I've opened it up to the internet.

**Pull request merged**